bika/lims/utils/analysisrequest.py

```diff
@@ -93,10 +93,22 @@ def create_analysisrequest(client, request, values, analyses=None,
     # Create sample partitions
     if not partitions:
         partitions = [{'services': service_uids}]
+
+    part_num = 0
+    if secondary:
+        # Always create new partitions if is a Secondary AR, cause it does
+        # not make sense to reuse the partitions used in a previous AR!
+        sparts = sample.getSamplePartitions()
+        prefix = sample.getId() + "-P"
+        for spart in sparts:
+            spartnum = int(spart.getId().split(prefix)[1])
+            if spartnum > part_num:
+                part_num = spartnum
+
     for n, partition in enumerate(partitions):
         # Calculate partition id
         partition_prefix = sample.getId() + "-P"
```

Comment:
I think this line is just repeat of the line # 102 ? Should we take 102 out of if condition and delete this line?